src/mono/wasm/wasm.proj

```diff
@@ -84,12 +84,29 @@
           AfterTargets="Build"
           DependsOnTargets="BuildPInvokeTable;BundleTimeZones">
 
-    <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/browser-wasm/native/lib ENABLE_ES6=$(WasmEnableES6)"
+    <PropertyGroup>
+      <_IcuLibdir>$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)/runtimes/browser-wasm/native/lib</_IcuLibdir>
+    </PropertyGroup>
+
+    <Exec Command="make -C $(MonoProjectRoot)wasm all SHELL=/bin/bash BINDIR=$(ArtifactsBinDir) MONO_BIN_DIR=$(MonoArtifactsPath) OBJDIR=$(ArtifactsObjDir) NATIVE_BIN_DIR=$(NativeBinDir) CONFIG=$(Configuration) PINVOKE_TABLE=$(WasmPInvokeTablePath) ICU_LIBDIR=$(_IcuLibdir) ENABLE_ES6=$(WasmEnableES6)"
           IgnoreStandardErrorWarningFormat="true" />
 
     <Copy SourceFiles="$(NativeBinDir)dotnet.js;
                        $(NativeBinDir)dotnet.wasm;
-                       $(NativeBinDir)dotnet.timezones.blat"
+                       $(NativeBinDir)dotnet.timezones.blat;
+                       $(MonoProjectRoot)wasm\runtime\driver.c;
+                       $(MonoProjectRoot)wasm\runtime\pinvoke.c;
+                       $(MonoProjectRoot)wasm\runtime\corebindings.c;
```

Comment:
Can this be precompiled?

Reply:
Yes, but they are pretty fast to compile, and driver.c needs still needs to be compiled because it has some conditional sections for aot/debug etc.

Reply:
I'm concerned about the long term reliability, cannot we rely on linker only to do the job?

Reply:
This is how driver.c was compiled in the old repo, and there were no problems with it. Both driver.c and pinvoke.c can include generated pinvoke/icall tables so they need to be compiled.

Reply:
Maybe we can move the conditional parts to a separate file, but it would be a lot of work.